plan.py

```diff
@@ -55,26 +157,48 @@ def expansion(node):
     # Convert mols to format for prediction
     # If the mol is in the starting set, ignore
     mols = [mol for mol in mols if mol not in starting_mols]
-    fprs = policies.fingerprint_mols(mols)
+    fprs = policies.fingerprint_mols(mols, expansion_net.X.shape[1])
+
+    if not args.use_openvino:
+        # Predict applicable rules
+        preds = sess.run(expansion_net.pred, feed_dict={
+            expansion_net.keep_prob: 1.,
+            expansion_net.X: fprs,
+            expansion_net.k: 5
+        })
+
+        indices = preds.indices[0]
+        values = preds.values[0]
+        np.save('expansion_preds_tf', [indices, values])
+
+    else:
+        # Get output nodes names
+        values_blob = list(iter(exp_net.outputs))[0]
+        indices_blob = list(iter(exp_net.outputs))[1]
 
-    # Predict applicable rules
-    preds = sess.run(expansion_net.pred_op, feed_dict={
-        expansion_net.keep_prob: 1.,
-        expansion_net.X: fprs,
-        expansion_net.k: 5
-    })
+        # Predict applicable rules
+        preds = exp_exec_net.infer(inputs={exp_inp_node_x_name: fprs})
+        
+        indices = preds[indices_blob][0]
+        values = preds[values_blob][0]
+        np.save('expansion_preds_ov', [indices, values])
+
+        top_k = namedtuple('TopKV2', 'values indices')
+        preds = top_k([values], [indices])
 
     # Generate children for reactants
     children = []
-    for mol, rule_idxs in zip(mols, preds):
+    for mol, rule_idxs in zip(mols, preds.indices):
         # State for children will
         # not include this mol
-        new_state = mols - {mol}
+
+        smol = set(mol)
+        new_state = [x for x in mols if x in smol]
 
         mol = Chem.MolFromSmiles(mol)
         for idx in rule_idxs:
             # Extract actual rule
-            rule = expansion_rules[idx]
+            rule = list(expansion_rules.keys())[list(expansion_rules.values()).index(idx)]
```

Comment:
Hard to understand this fix. Can you please comment?

Reply:
`rule` is a `dict`, where idx is value and rule is key, so we should get the key by value.

Reply:
Can you please show an example? Value of `expansion_rules` and one of the `idx` + `rule`

Reply:
`expansion_rules` is too large to show, `rule` + `idx`:  
```
('[C&H3:12]-[C:11]1(-[#7:10]2-[C:8](=[O:9])-[c:7]3:[c:2](-[N&H2:1]):[c:3]:[c:4]:[c:5]:[c:6]:3-[C:20]-2=[O:21])-[C:13]-[C:14]-[C:15](=[O:16])-[#7:17]-[C:18]-1=[O:19]>>[#7;+:1]-[c:2]1:[c:3]:[c:4]:[c:5]:[c:6]2:[c:7]:1-[C:8](=[O:9])-[#7:10](-[C:11]1(-[C&H3:12])-[C:13]-[C:14]-[C:15](=[O:16])-[#7:17]-[C:18]-1=[O:19])-[C:20]-2=[O:21]', 0)
```

Reply:
Can we just swap keys and values this way?
```python
expansion_rules = rules['expansion']
expansion_rules_new = {}
for rule, idx in expansion_rules.items():
    expansion_rules_new[idx] = rule
```